main.go

```diff
@@ -217,26 +219,25 @@ func createKubeClient(inCluster bool, apiserver string, kubeconfig string) (kube
 		}
 		glog.Infof("service account token present: %v", tokenPresent)
 		glog.Infof("service host: %s", config.Host)
-		if kubeClient, err = clientset.NewForConfig(config); err != nil {
-			return nil, err
-		}
 	} else {
 		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
 		// if you want to change the loading rules (which files in which order), you can do so here
 		loadingRules.ExplicitPath = kubeconfig
 		configOverrides := &clientcmd.ConfigOverrides{}
 		// if you want to change override values or bind them to flags, there are methods to help you
 		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
-		config, err := kubeConfig.ClientConfig()
+		config, err = kubeConfig.ClientConfig()
 		//config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
 		//config, err := clientcmd.DefaultClientConfig.ClientConfig()
 		if err != nil {
 			return nil, err
 		}
-		kubeClient, err = clientset.NewForConfig(config)
-		if err != nil {
-			return nil, err
-		}
+	}
+
+	config.ContentType = "application/vnd.kubernetes.protobuf"
```

Comment:
Use `AcceptContentTypes = "application/vnd.kubernetes.protobuf,application/json` instead

Reply:
Yes, of course. Now we're getting an error though in the node collector:

```
E0921 18:19:56.181928    1414 streamwatcher.go:109] Unable to decode an event from the watch stream: invalid character '\x00' looking for beginning of value
E0921 18:19:56.186126    1414 streamwatcher.go:109] Unable to decode an event from the watch stream: invalid character '\x00' looking for beginning of value
W0921 18:19:56.186202    1414 reflector.go:323] k8s.io/kube-state-metrics/collectors/node.go:140: watch of *v1.Node ended with: very short watch
```

Any idea?

Reply:
seems like encoder and decoder mismatch

Reply:
@smarterclayton What is the difference between `application/vnd.kubernetes.protobuf` and `application/vnd.kubernetes.protobuf,application/json`.

Reply:
According to [client-go #76](https://github.com/kubernetes/client-go/issues/76), protobuf is not the default data type and may change in the future.

@smarterclayton @brancz Should we make this optional through command line argument.

Reply:
@andyxning My guess would be that `application/vnd.kubernetes.protobuf,application/json` is content negotiation, that when an endpoint doesn't support proto, then it responds in json and sets the content type header appropriately.

I don't think we should make it optional, I don't see a reason why people would want to choose.

Reply:
@brancz If apiserver supports content negotiation, it should be fine to use this and we do not need to make protobuf optional since client-go should fall back to json data type once protobuf is not supported.

Reply:
@brancz Thanks for pointing this out. It should be according to [rfc 2296](https://tools.ietf.org/html/rfc2296).

Reply:
I have tested this PR with Kubernetes 1.6.3. Everything is fine. However, in order to debug i suggest we get and log the actual content type used between apiserver and kube-state-metrics.

Reply:
> @brancz If apiserver supports content negotiation, it should be fine to use this and we do not need to make protobuf optional since client-go should fall back to json data type once protobuf is not supported.

@andyxning I used crd typed client and  set `application/vnd.kubernetes.protobuf,application/json` is content negotiation. but the client is abnormal.

```
Failed to watch xxxxxx: failed to list *v1alpha1.DuplicateSet: object *unstructured.UnstructuredList does not implement the protobuf marshalling interface and cannot be encoded to a protobuf message
```

https://kubernetes.io/docs/reference/using-api/api-concepts/#alternate-representations-of-resources